File: Beginner/Cats and Dogs (CATSDOGS)/catsdogs.py
t = int(input())

while t:
    C, D, L = map(int, input().split())

    M = C*4 + D*4
    m = D*4
    if C > 2*D:
        m += (C-2*D)*4
    if m <= L <= M and L % 4 == 0:
        print('yes')
    else:
        print('no')

    t = t-1
        

File: Beginner/Cats and Dogs (CATSDOGS)/catsdogs.cpp
#include <iostream>
using namespace std;

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        long int cat, dog, leg;
        cin >> cat >> dog >> leg;
        long int max_leg = 4 * (cat + dog);
        long int min_leg = 4 * dog;
        if (2 * dog < cat)
        {
            min_leg += (cat - (2 * dog)) * 4;
        }
        if (leg <= max_leg && leg >= min_leg && leg % 4 == 0)
            cout << "yes\n";
        else
            cout << "no\n";
    }
    return 0;
}
